src/networking.c

```diff
@@ -117,6 +118,65 @@ int authRequired(client *c) {
     return auth_required;
 }
 
+sds getAllClientMetaFields(dict *meta) {
+    dictIterator *di = dictGetIterator(meta);
+    dictEntry *de;
+    sds result = sdscatfmt(sdsempty(),
+        "%%%i\n",
+        (int) dictSize(meta)
+    );
+
+    while((de = dictNext(di)) != NULL) {
+        sds key = dictGetKey(de);
+        sds value = dictGetVal(de);
+
+        result = sdscatfmt(result,
+            "+%s\n:%s\n",
+            key,
+            value
+        );
+    }
+    dictReleaseIterator(di);
+
+    return result;
+}
+
+sds getClientMetaFields(client *c) {
+    int i;
+    int count = 0;
+    sds partialResult = sdsempty();
+
+    for (i = 2; i < c->argc; i++) {
+        dictEntry *de;
+        de = dictFind(c->meta, c->argv[i]->ptr);
+        if (de == NULL) {
+            continue;
+        };
+        sds value = dictGetVal(de);
+        
+        partialResult = sdscatfmt(partialResult,
+            "+%s\n:%s\n",
+            c->argv[i]->ptr,
+            value
+        );
+        count++;
+    }
```

Comment:
Seems like I didn't do a good job explaining the correct way to do this. It looks like you're trying to build up the map result yourself. There is a bunch of helper methods for this.
```suggestion
    void *ufields = addReplyDeferredLen(c);
    for (i = 2; i < c->argc; i++) {
        dictEntry *de;
        de = dictFind(c->meta, c->argv[i]->ptr);
        if (de == NULL) {
            continue;
        };
        sds value = dictGetVal(de);
        
        addReply(c, c->argv[i]);
        addReplyBulkCBuffer(c, value, sdslen(value));
        count++;
    }
    setDeferredMapLen(c, ufields, count);
```

Reply:
thank you @madolson, I follow your suggestion, I only got an error when using `addReply(c,c->argv[i]);` so I just changed to use addReplyBulkCBuffer passing the key string